test/simulator/misc/choice-parser.js

```diff
@@ -80,6 +80,21 @@ describe('Choice parser', function () {
 		});
 
 		describe('Doubles/Triples', function () {
+			it('should allow mega evolving and targeting in the same move in either order', function () {
+				battle = common.createBattle({gameType: 'doubles'});
+				battle.join('p1', 'Guest 1', 1, [
+					{species: "Gengar", ability: 'cursedbody', item: 'gengarite', moves: ['shadowball']},
+					{species: "Koffing", ability: 'levitate', moves: ['smog']},
+				]);
+				battle.join('p2', 'Guest 2', 1, [
+					{species: "Blaziken", ability: 'speedboost', item: 'blazikenite', moves: ['blazekick']},
+					{species: "Aggron", ability: 'sturdy', moves: ['irondefense']},
+				]);
+
+				assert.ok(battle.choose('p1', `move 1 1 mega, move smog 1`));
+				assert.ok(battle.choose('p2', `move 1 mega 2, move irondefense`));
+			});
+
```

Comment:
This test was placed under ``Switch requests > Doubles/Triples``. It should be under ``Move requests > Generic``